openff/toolkit/topology/molecule.py

```diff
@@ -161,7 +163,7 @@ def to_dict(self):
         raise NotImplementedError()  # TODO
 
     @classmethod
-    def from_dict(cls, d):
+    def from_dict(cls: type[P], d: dict) -> P:
```

Comment:
```suggestion
    def from_dict(cls: Type[P], d: dict) -> P:
```

I think? If `type` works, we should use it instead of `Type` elsewhere in this module.